hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -215,12 +213,15 @@ getTxId tx = Ledger.TxId $ SafeHash.hashAnnotated (body tx)
 -- | This are all the error that can happen during coverFee.
 data ErrCoverFee
   = ErrNotEnoughFunds ChangeError
-  | ErrNoFuelUTxOFound
+  | ErrNoUTxOFound
   | ErrUnknownInput {input :: TxIn}
   | ErrScriptExecutionFailed {scriptFailure :: (RdmrPtr, TransactionScriptFailure LedgerEra)}
   | ErrTranslationError (TranslationError StandardCrypto)
   deriving (Show)
 
+-- TODO: provide To/From JSON instances for ErrCoverFee so that we can reuse
+-- them in InternalWalletError instead of putting just the Text for reason.
+--
```

Comment:
I would strongly advise against this. The `ErrCoverFee` type is not intended to be used in an interface and it's typically a good idea to constrain de-/serialization to the "edges" of a system. We have `PostTxError` already which can/should be used for this. Or even better, one or more dedicated error types in `Hydra.Api...` with serialization instances should be used.